### PR TITLE
Remap alpha channel of baseColorMap

### PR DESCRIPTION
**Pull request by skhiat**
Ref:
https://github.com/Unity-Technologies/Graphics/pull/6201
 (already reviewed)

Remap Transparent Channel of the BaseColor/Color texture for {Lit, Unlit, LayeredLit, LitTesselation, LayeredLitTesselation}
Lit:
image

Unlit:
image

LayeredLit:
image

LitTesselation:
image

LayeredLitTesselation:
image

### CODE REVIEW

**Review by Vic-Cooper — APPROVED** [review]
These docs changes are consistent with the existing docs, so no changes needed!

**Review by kecho — APPROVED** [review]
Lets remerge, rerun yamato. Looks good from my side

File: com.unity.render-pipelines.high-definition/Runtime/Material/Unlit/UnlitData.hlsl [hunk 1]
```diff
@@ -30,7 +30,7 @@ void GetSurfaceAndBuiltinData(FragInputs input, float3 V, inout PositionInputs p
 
     // Builtin Data
     ZERO_BUILTIN_INITIALIZE(builtinData); // No call to InitBuiltinData as we don't have any lighting
-    builtinData.opacity = alpha;
+    builtinData.opacity = lerp(_AlphaRemapMin, _AlphaRemapMax, alpha);
```
Comment (sebastienlagarde) [inline com.unity.render-pipelines.high-definition/Runtime/Material/Unlit/UnlitData.hlsl:33]: This is not correct, alpha need to be modify at the source here, in 
float alpha = SAMPLE_TEXTURE2D(_UnlitColorMap, sampler_UnlitColorMap, unlitColorMapUv).a * _UnlitColor.a;
=>
float alpha = SAMPLE_TEXTURE2D(_UnlitColorMap, sampler_UnlitColorMap, unlitColorMapUv).a * _UnlitColor.a;
alpha = lerp(_AlphaRemapMin, _AlphaRemapMax, alpha);

as it is used in the alpha test after that GENERIC_ALPHA_TEST(alpha, _AlphaCutoff); . Same is true for lit shader

**Review by sebastienlagarde — CHANGES_REQUESTED** [review]
This don't work correctly with alpha test. See comment in unlit.hlsl

Also you haven't udpate the LitData.hlsl
here: float alphaValue = SAMPLE_UVMAPPING_TEXTURE2D(_BaseColorMap, sampler_BaseColorMap, layerTexCoord.base).a * _BaseColor.a;

**Comment by sebastienlagarde** [comment]
For QA: think to check the slider with alpha test

File: com.unity.render-pipelines.high-definition/Runtime/Material/Unlit/UnlitData.hlsl (hunk 1 above)
Comment (skhiat) [reply com.unity.render-pipelines.high-definition/Runtime/Material/Unlit/UnlitData.hlsl:33]: Open question:
That should be:
float alpha = SAMPLE_TEXTURE2D(_UnlitColorMap, sampler_UnlitColorMap, unlitColorMapUv).a * _UnlitColor.a;
alpha = lerp(_AlphaRemapMin, _AlphaRemapMax, alpha);
Or
float alpha = SAMPLE_TEXTURE2D(_UnlitColorMap, sampler_UnlitColorMap, unlitColorMapUv).a
alpha = lerp(_AlphaRemapMin, _AlphaRemapMax, alpha);
alpha *= _UnlitColor.a;

The remap should be on data only?

**Comment by RemyUnity** [comment]
Should we consider adding a graphic test for this ?

**Review by RemyUnity — CHANGES_REQUESTED** [review]
The changes are ok for me, I roughly tested it on my side but @remi-chapelain already covered all materials in the [original PR](https://github.com/Unity-Technologies/Graphics/pull/6201).
Now it just needs coverage graphic tests (or maybe can be merged already if we don't forget to do it later on @sebastienlagarde )

**Review by sebastienlagarde — APPROVED** [review]

**Pull request merged**